stdlib/array.gr

```diff
@@ -121,15 +121,63 @@ provide let init = (length: Number, fn: Number => a) => {
   let mut index = 0n
   for (let mut i = 0n; i < byteLength; i += 4n) {
     WasmI32.store(
-      array + i,
+      array,
       Memory.incRef(WasmI32.fromGrain(fn(tagSimpleNumber(index)))),
-      _ARRAY_START_OFFSET
+      _ARRAY_START_OFFSET + i
     )
     index += 1n
   }
   WasmI32.toGrain(array): Array<a>
 }
 
+/**
+ * Creates a new array of the specified length where each element is
+ * initialized with the result of an initializer function. The initializer
+ * is called with the index of each array element. If the initializer function
+ * returns `Err(_)`, the array creation is stopped and the error is returned.
```

Comment:
```suggestion
 * Creates a new array where each element is initialized with the `Ok`
 * value result of an initializer function. The initializer is called with the
 * index of each element. Returns the new array if all calls to the
 * initializer succeed or the first error otherwise.
```